### PR TITLE
fix coverify warning

### PR DESCRIPTION
**Pull request by zhuyanlinzyl**
## Summary

fix coverify warning

### CODE REVIEW

**Review by xiaoxiang781216 — APPROVED** [review]

File: sched/sched/sched_note.c [hunk 1]
```diff
@@ -494,7 +494,7 @@ void sched_note_start(FAR struct tcb_s *tcb)
   namelen = strlen(tcb->name);
 
   DEBUGASSERT(namelen <= CONFIG_TASK_NAME_SIZE);
-  strncpy(note.nsa_name, tcb->name, CONFIG_TASK_NAME_SIZE + 1);
+  strlcpy(note.nsa_name, tcb->name, CONFIG_TASK_NAME_SIZE + 1);
```
Comment (pkarashchenko) [inline sched/sched/sched_note.c:497]: Should we remove +1 now?

Comment (xiaoxiang781216) [reply sched/sched/sched_note.c:497]: no, since note.nsa_name contain `CONFIG_TASK_NAME_SIZE + 1`'s char,

Comment (pkarashchenko) [reply sched/sched/sched_note.c:497]: Then maybe switch to sizeof?

File: sched/module/mod_insmod.c [hunk 2]
```diff
@@ -198,9 +198,11 @@ FAR void *insmod(FAR const char *filename, FAR const char *modname)
       goto errout_with_loadinfo;
     }
 
+#ifdef HAVE_MODLIB_NAMES
   /* Save the module name in the registry entry */
 
   strlcpy(modp->modname, modname, MODLIB_NAMEMAX);
```
Comment (pkarashchenko) [inline sched/module/mod_insmod.c:204]:
```suggestion
  strlcpy(modp->modname, modname, sizeof(modp->modname));
```

**Review by pkarashchenko — APPROVED** [review]
One minor optional change

**Review by pkarashchenko — APPROVED** [review]
LGTM!

**Pull request merged**